app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class GroupsController < ApplicationController
+  def index
+    redirect_to root_path unless current_user
```

Comment:
this row should be removed because we `authenticate_user` action in `application_controller.rb`